packages/react/src/componentsV2/heliopay-container/index.tsx

```diff
@@ -0,0 +1,255 @@
+    setResult(event);
+    setShowLoadingModal(false);
+  };
+
+  const submitPayment = async ({
+    amount,
+    currency,
+    quantity,
+    customerDetails,
+    productDetails,
+  }: {
+    amount: number;
+    currency: Currency;
+    quantity: number;
+    customerDetails?: CustomerDetails;
+    productDetails?: ProductDetails;
+  }) => {
+    if (helioProvider && currency?.symbol != null) {
+      onStartPayment?.();
+      setShowLoadingModal(true);
+      setShowFormModal(false);
+      const recipient = paymentDetails?.wallet?.publicKey;
+      const { symbol } = getCurrency(currency.symbol);
+      if (symbol == null) throw new Error('Unknown currency symbol');
+      const payload = {
+        anchorProvider: helioProvider,
+        recipientPK: recipient,
+        symbol,
+        amount: amount * (quantity || 1),
+        paymentRequestId,
+        onSuccess: handleSuccessPayment,
+        onError: handleErrorPayment,
+        onPending,
+        customerDetails,
+        quantity: Number(quantity),
+        productDetails,
+        splitRevenue: paymentDetails?.features?.splitRevenue,
+        splitWallets: paymentDetails?.splitWallets,
+        wallet: wallet as AnchorWallet,
+        connection: connectionProvider.connection,
+        rateToken: dynamicRateToken,
+        cluster,
+        canSwapTokens: paymentDetails?.features.canSwapTokens,
+        swapRouteToken: tokenSwapQuote?.routeTokenString,
+      };
+
+      try {
+        await HelioSDK.paylinkService.handleTransaction(payload);
+      } catch (error) {
+        handleErrorPayment({
+          errorMessage: String(error),
+        });
+      }
+    }
+  };
+
+  return (
+    <StyledWrapper>
+      {!result ? (
+        <>
+          <StyledRow>
+            <StyledLeft>
+              {wallet ? (
+                <div>
+                  <Button
+                    onClick={() => {
+                      if (
+                        isCustomerDetailsRequired ||
+                        supportedCurrencies?.length
+                      ) {
+                        setShowFormModal(true);
+                      } else {
+                        submitPayment({
+                          amount: paymentDetails?.normalizedPrice,
+                          quantity: 1,
+                          customerDetails: undefined,
+                          currency: paymentDetails?.currency,
+                        });
+                      }
+                    }}
+                    disabled={!paymentDetails?.id}
+                  >
+                    {payButtonTitle}
+                  </Button>
+                </div>
+              ) : (
+                <ConnectButton />
+              )}
+            </StyledLeft>
+            <StyledRight>
+              Powered by
+              <StyledLogoContainer>
+                <StyledLogo>
+                  <HelioLogoGray />
+                </StyledLogo>
+                {cluster === ClusterType.Devnet && (
```

Comment:
FYI we updated the name 'cluster' to generalNetwork on dashboard as 'cluster' is a Solana specific term. I kept it backwards compatible though and made sure we still use cluster when generating React button code, but one to remember.